packages/react-native-prebuilt/src/index.ts

```diff
@@ -0,0 +1,329 @@
+
+async function nativeBabelFlowTransform(input: string) {
+  return await new Promise<string>((res, rej) => {
+    babel.transform(
+      input,
+      {
+        presets: ['module:metro-react-native-babel-preset'],
+      },
+      (err: any, result) => {
+        if (!result || err) rej(err || 'no res')
+        res(result!.code!)
+      }
+    )
+  })
+}
+
+const external = ['react', 'react/jsx-runtime', 'react/jsx-dev-runtime']
+const outPath = './dist/react-native.js'
+const reactOutPath = './dist/react.js'
+const reactJsxOutPath = './dist/react-jsx-runtime.js'
+
+export async function buildAll() {
+  console.info(`Prebuilding React Native (one time cost...)`)
+  await Promise.all([buildReactJSX(), buildReact(), buildReactNative()])
+}
+
+export async function buildReactJSX(options: BuildOptions = {}) {
+  return build({
+    bundle: true,
+    entryPoints: [require.resolve('react/jsx-dev-runtime')],
+    outfile: reactJsxOutPath,
+    format: 'cjs',
+    target: 'node16',
+    jsx: 'transform',
+    jsxFactory: 'react',
+    allowOverwrite: true,
+    platform: 'node',
+    define: {
+      __DEV__: 'true',
+      'process.env.NODE_ENV': `"development"`,
+    },
+    external,
+    logLevel: 'warning',
+    ...options,
+  }).then(async () => {
+    // manual force exports
+    const bundled = await readFile(reactJsxOutPath, 'utf-8')
+    const outCode = `
+    const run = () => {
+      ${bundled.replace(
+        `module.exports = require_react_jsx_dev_runtime_development();`,
+        `return require_react_jsx_dev_runtime_development();`
+      )}
+    }
+    const __mod__ = run()
+    ${['jsx', 'jsxs', 'jsxDEV', 'Fragment']
+      .map((n) => `export const ${n} = __mod__.${n} || __mod__.jsx || __mod__.jsxDEV`)
+      .join('\n')}
+    `
+    await writeFile(reactJsxOutPath, outCode)
+  })
+}
+
+export async function buildReact(options: BuildOptions = {}) {
+  return build({
+    bundle: true,
+    entryPoints: [require.resolve('react')],
+    outfile: reactOutPath,
+    format: 'cjs',
+    target: 'node16',
+    jsx: 'transform',
+    jsxFactory: 'react',
+    allowOverwrite: true,
+    platform: 'node',
+    define: {
+      __DEV__: 'true',
+      'process.env.NODE_ENV': `"development"`,
+    },
+    logLevel: 'warning',
+    external,
+    ...options,
+  }).then(async () => {
+    // manual force exports
+    const bundled = await readFile(reactOutPath, 'utf-8')
+    const outCode = `
+    const run = () => {
+      ${bundled.replace(
+        `module.exports = require_react_development();`,
+        `return require_react_development();`
+      )}
+    }
+    const __mod__ = run()
+    ${RExports.map((n) => `export const ${n} = __mod__.${n}`).join('\n')}
+    export default __mod__
+    `
+    await writeFile(reactOutPath, outCode)
+  })
+}
+
+export async function buildReactNative(options: BuildOptions = {}) {
+  return build({
+    bundle: true,
+    entryPoints: [require.resolve('react-native')],
```

Comment:
shouldn't we resolve this path from the project's root? since right now `react-native-prebuilt` has `react-native` dependency, so it will pick it instead of the project's one.

Reply:
oh yea definitely, good catch

Reply:
i think i am in `create` now but i'll doubel check it